dashboard/test/models/user_test.rb

```diff
@@ -1563,6 +1563,18 @@ def track_progress(user_id, script_level, result, pairings: nil)
     assert_equal [], teacher.reload.permissions
   end
 
+  test 'assign_course_as_facilitator assigns course to facilitator' do
+    facilitator = create :facilitator
+    facilitator.course_as_facilitator = Pd::Workshop::COURSES.last
+    assert facilitator.courses_as_facilitator.where(course: Pd::Workshop::COURSES.last).any?
+  end
+
+  test 'delete_course_as_facilitator removes facilitator course' do
+    facilitator = create :facilitator
+    facilitator.delete_course_as_facilitator Pd::Workshop::COURSES.first
+    refute facilitator.courses_as_facilitator.where(course: Pd::Workshop::COURSES.first).any?
+  end
```

Comment:
I experimented with adding a trait or a transient attribute, but then realized I could use the existing pd_course_facilitator factory and the facilitator it belongs to in the delete test.

I fixed the `assert_difference` test by modifying `assert_difference 'Pd::CourseFacilitator'` to `assert_difference 'Pd::CourseFacilitator.count'`